File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Lithuania.m
//
//  UIImage+Lithuania.m
//  CountriesFlags
//

#import "Countries/UIImage+Lithuania.h"

@implementation UIImage (Lithuania)

+ (UIImage * _Nullable)flagLithuaniaWithSize:(CGSize)size {
    UIColor* colorYellow = [UIColor colorWithRed:253.0f/255.0f
                                           green:185.0f/255.0f
                                            blue:19.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:106.0f/255.0f
                                           blue:68.0f/255.0f
                                          alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:193.0f/255.0f
                                        green:39.0f/255.0f
                                         blue:45.0f/255.0f
                                        alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorYellow setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f, size.width, size.height / 3.0f));
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f * 2.0f, size.width, size.height / 3.0f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Lithuania.swift
//
//  UIImage+Lithuania.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Lithuania "wikipedia"
     
     Generates the flag of Lithuania.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Lithuania.
     */
    internal static func flagLithuania(with size: CGSize) -> UIImage {
        let colorYellow = UIColor(red: 253.0 / 255.0,
                                  green: 185.0 / 255.0,
                                  blue: 19.0 / 255.0,
                                  alpha: 1.0)
        let colorGreen = UIColor(red: 0.0,
                                 green: 106.0 / 255.0,
                                 blue: 68.0 / 255.0,
                                 alpha: 1.0)
        let colorRed = UIColor(red: 193.0 / 255.0,
                               green: 39.0 / 255.0,
                               blue: 45.0 / 255.0,
                               alpha: 1.0)
        let lineSize = CGSize(width: size.width, height: size.height / 3.0)

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorYellow.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0), size: lineSize))

        colorRed.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0 * 2.0), size: lineSize))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image ?? UIImage()
    }
}

#endif
